public/components/ui_action_context_menu.tsx

```diff
@@ -75,14 +107,13 @@ export const ActionContextMenu = (props: Props) => {
     [actionContext.datasetId, actionContext.datasetType, actionContext.dataSourceId]
   );
 
-  // The action button should be not displayed when there is no action and result summary disabled.
-  if (actionsRef.current.length === 0 && !resultSummaryEnabled) {
+  // The action button should be not displayed when there is no action and result summary disabled or there is no data2Summary agent
+  if (!isSummaryAgentAvailable || (actionsRef.current.length === 0 && !resultSummaryEnabled)) {
     return null;
   }
 
-  // The action button should be disabled when context menu has no item and result summary disabled.
+  // The action button should be disabled when context menu has no item and result summary disabled
   const actionDisabled = (panels.value?.[0]?.items ?? []).length === 0 && !resultSummaryEnabled;
```

Comment:
```suggestion
  const actionDisabled = (panels.value?.[0]?.items ?? []).length === 0 && !resultSummaryEnabled && !isSummaryAgentAvailable;
```
Do we need to add check on `isSummaryAgentAvailable`?

Reply:
I add this check at first, but if there isn't agent, should we just hide the button instead of disabling it?

Reply:
I think it is OK that we check on both places, but sure it is good enough if we will hide the button if no summary agent is available on target data source.

Reply:
update~ and thanks for all the comments